widget/tree.go

```diff
@@ -186,6 +188,40 @@ func (t *Tree) Selection() string {
 // SetSelection updates the current selection to the node with the given Unique ID.
 func (t *Tree) SetSelection(uid string) {
 	t.selected = uid
+	if t.selected != TreeNoSelection && t.scroller != nil {
+		var found bool
+		var y int
+		var size fyne.Size
+		t.walkAll(func(uid string, branch bool, depth int) {
+			m := t.leafMinSize
+			if branch {
+				m = t.branchMinSize
+			}
+			if uid == t.selected {
+				found = true
+				size = m
+			} else if !found {
+				// Root node is not rendered unless it has been customized
+				if t.Root == "" && uid == "" {
+					// This is root node, skip
+					return
+				}
+				// If this is not the first item, add a divider
+				if y > 0 {
+					y += treeDividerHeight
+				}
+
+				y += m.Height
+			}
+		})
+		if y < t.scroller.Offset.Y {
+			t.scroller.Offset.Y = y
+		} else if y+size.Height > t.scroller.Offset.Y+t.scroller.Size().Height {
+			t.scroller.Offset.Y = y + size.Height - t.scroller.Size().Height
+		}
+		t.scroller.onOffsetChanged()
+		// TODO Setting a node as selected should open all parents if they aren't already
```

Comment:
For the future I think there may be a way that we can do this in the same pass that we are doing walkall - I.E.  Open the parents as we walk into them looking for selection and then close them if the item isn't there - other wise if the item is found you can leave the parents open.